client-src/default/utils/updatePublicPath.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+/* global __webpack_public_path__ */
+
+const url = require('url');
+const getUrlParts = require('./getUrlParts');
+
+function updatePublicPath(resourceQuery) {
+  const urlParts = getUrlParts(resourceQuery);
+
+  let publicPath = urlParts.publicPath;
+
+  // this tests if publicPath is relative, and makes it an absolute url if it is
+  // will match something of the form '/.../', or simply '/'
+  if (/^\/.*\/$|^\/$/.test(publicPath)) {
```

Comment:
Can we encapsulate this in `isRelativeUrl` (maybe using other repository for this check)?